tracing-opentelemetry/src/main/java/io/micronaut/tracing/processing/WithSpanAnnotationMapper.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.tracing.processing;
+
+import io.micronaut.aop.InterceptorBinding;
+import io.micronaut.aop.InterceptorKind;
+import io.micronaut.core.annotation.AnnotationValue;
+import io.micronaut.inject.annotation.TypedAnnotationMapper;
+import io.micronaut.inject.visitor.VisitorContext;
+import io.micronaut.tracing.annotation.ContinueSpan;
+import io.opentelemetry.extension.annotations.WithSpan;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Adds interceptor binding to {@link WithSpan} and adds {@link ContinueSpan} to it.
+ *
+ * @author Nemanja Mikic
+ */
+public class WithSpanAnnotationMapper implements TypedAnnotationMapper<WithSpan> {
+
+    @Override
+    public List<AnnotationValue<?>> map(AnnotationValue<WithSpan> annotation, VisitorContext visitorContext) {
+        AnnotationValue<InterceptorBinding> interceptorBinding = AnnotationValue.builder(InterceptorBinding.class)
+            .value(ContinueSpan.class)
```

Comment:
Looks like it's not correct to map `WithSpan` to `ContinueSpan` annotation, it is mapped to `NewSpan` annotation instead, based on its description from javadoc:

```
This annotation marks that an execution of this method or constructor should result in a new io.opentelemetry.api.trace.Span.
Application developers can use this annotation to signal OpenTelemetry auto-instrumentation that a new span should be created whenever marked method is executed.
```

Reply:
I have changed `WithSpan` to match the` NewSpan`. Grpc test added, and I don't know how to fix test that is falling on ci / cd because localy it is passing: https://ge.micronaut.io/s/53d2qu5xuicqy. The issue is in  mapping/transforming annotations.